controllers/btpoperator_controller_test.go

```diff
@@ -500,6 +516,313 @@ var _ = Describe("BTP Operator controller", Ordered, func() {
+				Expect(bytes.Equal(afterCaSecretCert, beforeCaSecretCert))
+				ensureCorrectState()
+			})
+		})
+
+		When("webhook certificate and is signed by different CA certificate", func() {
+			BeforeAll(func() {
+				certBeforeEach(nil)
+			})
+			AfterAll(func() {
+				certAfterEach()
+			})
+			It("CA certificate and webhook certificate are fully regenerated", func() {
+				newCaCertificate, newCaPrivateKey, err := certs.GenerateSelfSignedCertificate(time.Now().Add(CaCertificateExpiration))
+				Expect(err).To(BeNil())
+
+				newWebhookCertificate, newWebhookPrivateKey, err := certs.GenerateSignedCertificate(time.Now().Add(WebhookCertificateExpiration), newCaCertificate, newCaPrivateKey)
+				newWebhookCertificateStructured, err := structToByteArray(newWebhookPrivateKey)
+				Expect(err).To(BeNil())
+
+				beforeCaSecret := getSecret(CaSecret)
+				beforeWebhookSecret := getSecret(WebhookSecret)
+
+				webhookCertSecret := getSecret(WebhookSecret)
+				replaceSecretData(webhookCertSecret, reconciler.buildKeyNameWithExtension(WebhookSecretDataPrefix, CertificatePostfix), newWebhookCertificate, reconciler.buildKeyNameWithExtension(WebhookSecretDataPrefix, RsaKeyPostfix), newWebhookCertificateStructured)
+				ensureReconcilationQueueIsEmpty()
+
+				currentCaSecret := getSecret(CaSecret)
+				currentCaCert, ok := currentCaSecret.Data[reconciler.buildKeyNameWithExtension(CaSecretDataPrefix, CertificatePostfix)]
+				Expect(ok).To(BeTrue())
+				beforeCaCert, ok := beforeCaSecret.Data[reconciler.buildKeyNameWithExtension(CaSecretDataPrefix, CertificatePostfix)]
+				Expect(ok).To(BeTrue())
+				Expect(!bytes.Equal(currentCaCert, beforeCaCert))
+
+				currentWebhookSecret := getSecret(WebhookSecret)
+				currentWebhookCert, ok := currentWebhookSecret.Data[reconciler.buildKeyNameWithExtension(WebhookSecretDataPrefix, CertificatePostfix)]
+				Expect(ok).To(BeTrue())
+				beforeWebhookCert, ok := beforeWebhookSecret.Data[reconciler.buildKeyNameWithExtension(WebhookSecretDataPrefix, CertificatePostfix)]
+				Expect(ok).To(BeTrue())
+				Expect(!bytes.Equal(currentWebhookCert, beforeWebhookCert))
+				Expect(!bytes.Equal(currentWebhookCert, newWebhookCertificate))
+
+				ensureCorrectState()
+			})
+		})
+
+		When("webhook certificate expires", func() {
+			fakeTime := 30.0
+			fakeExpiration := 10.0
+			BeforeAll(func() {
+				timeOpts := &certificationsTimeOpts{
+					CaCertificateExpiration: CaCertificateExpiration,
+					WebhookCertExpiration:   time.Second * time.Duration(fakeTime),
+					ExpirationBoundary:      time.Second * time.Duration(fakeExpiration),
+				}
+				certBeforeEach(timeOpts)
+			})
+			AfterAll(func() {
+				certAfterEach()
+			})
+
+			It("CA certificate is not changed, webhook certificate is regenerated", func() {
+				caSecretBeforeExpiration := getSecret(CaSecret)
+				webhookSecretBeforeExpiration := getSecret(WebhookSecret)
+				Expect(checkHowManySecondsToExpiration(WebhookSecret) <= fakeTime).To(BeTrue())
```

Comment:
you can use gomega numerical matcher: https://onsi.github.io/gomega/#working-with-numbers-and-times